spring-integration-core/src/main/java/org/springframework/integration/config/xml/PointToPointChannelParser.java

```diff
@@ -55,14 +55,19 @@ protected BeanDefinitionBuilder buildBeanDefinition(Element element, ParserConte
 			builder = BeanDefinitionBuilder.genericBeanDefinition(QueueChannel.class);
 			boolean hasStoreRef = this.parseStoreRef(builder, queueElement, channel, false);
 			boolean hasQueueRef = this.parseQueueRef(builder, queueElement);
-			if (!hasStoreRef) {
+			if (!hasStoreRef || !hasQueueRef) {
 				boolean hasCapacity = this.parseQueueCapacity(builder, queueElement);
 				if (hasCapacity && hasQueueRef) {
 					parserContext.getReaderContext().error(
 							"The 'capacity' attribute is not allowed" + " when providing a 'ref' to a custom queue.",
 							element);
 				}
-			}
+				if (hasCapacity && hasStoreRef) {
+					parserContext.getReaderContext().error(
+							"The 'capacity' attribute is not allowed" + " when providing a 'message-store' to a custom MessageGroupStore.",
```

Comment:
Minor. Just for future PRs to save some time: https://github.com/spring-projects/spring-integration/wiki/Spring-Integration-Framework-Code-Style
We prefer line length no more than 120 symbols.